extensions/resteasy/runtime/src/main/java/io/quarkus/resteasy/runtime/standalone/VertxBufferImpl.java

```diff
@@ -0,0 +1,490 @@
+package io.quarkus.resteasy.runtime.standalone;
+
+import java.nio.ByteBuffer;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.util.Objects;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.Unpooled;
+import io.netty.util.CharsetUtil;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.impl.Arguments;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+
+public class VertxBufferImpl implements Buffer {
```

Comment:
you can directly create a `Buffer` from a `ByteBuf` with `Buffer.buffer(byteBuf)`

Reply:
do you need this to avoid `ByteBuf` duplication ?

Reply:
@vietj BufferImpl(ByteBuf) wraps it in an unreleasableBuffer

Reply:
right